### PR TITLE
feat: logs streamed through NATS

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/executor/containerexecutor/containerexecutor.go [hunk 1]
```diff
@@ -305,11 +308,18 @@ func (c *ContainerExecutor) updateResultsFromPod(
...
-	// wait for pod
-	l.Debug("poll immediate waiting for executor pod")
+	// wait for pod to be loggable
 	if err = wait.PollImmediate(pollInterval, c.podStartTimeout, executor.IsPodLoggable(ctx, c.clientSet, executorPod.Name, c.namespace)); err != nil {
 		l.Errorw("waiting for executor pod started error", "error", err)
-	} else if err = wait.PollImmediate(pollInterval, pollTimeout, executor.IsPodReady(ctx, c.clientSet, executorPod.Name, c.namespace)); err != nil {
+	}
+
+	// push logs to stream
+	l.Debug("starting pod logs stream proxy")
```
Comment (vsukhin) [inline pkg/executor/containerexecutor/containerexecutor.go:317]: should we do log streaming if we had error in above PollImmediate call?

File: pkg/executor/stream/logs.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	for _, container := range pod.Spec.InitContainers {
+		containers = append(containers, container.Name)
+	}
+
+	for _, container := range pod.Spec.Containers {
+		containers = append(containers, container.Name)
+	}
+
+	go func() {
+		for _, container := range containers {
```
[the remark this answers is not in the document]
Comment (vsukhin) [reply pkg/executor/stream/logs.go:36]: technically, in our case, we have 1 init container (executed beofre main) and 1 main container, so this loop is fine. but iif use sidecars, then it should be done in parallel

**Review by vsukhin — APPROVED** [review]
didn't use jetstream, but looks good, one small comment

File: pkg/executor/stream/logs.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (exu) [reply pkg/executor/stream/logs.go:36]: The final idea is to push logs to some API endpoint - so it'll be not needed at all

File: pkg/executor/containerexecutor/containerexecutor.go (hunk 1 above)
Comment (exu) [reply pkg/executor/containerexecutor/containerexecutor.go:317]: hmm - good point

File: pkg/executor/stream/logs.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (exu) [reply pkg/executor/stream/logs.go:36]: will it result in mixed logs then? maybe we would need to distinguish them ?

**Pull request closed without merging**